### PR TITLE
DS Foundational, + App Core styles & Positions + Base layout revamp + Agreement Layout revamp

### PR DESCRIPTION
**Pull request by gordo-labs**
Refactor of tailwind configs to be loaded one time with a centralised theme.

Set of foundational styles fitting the Figma structures.
Naming (WIP)

Core App Grid Structure Refactor

Positions Refactor & Styling

 The proper heading font failing.
 Adapt layout for listings.
Finalised a stable middle point to continue works from here
 The responsiveness built for new layout and working partially for lists layouts.
Finalised a stable middle point to continue works from here
 Built the No Action options and final component
 Fixed broken position typo
 I need to deep check
 Not found and loading state new components up
 Added illustration / icons library management in ui-components to automatically export as React components custom svgs
 And fixed navigation for judges dispute accessing

### CODE REVIEW

**Comment by xPi2** [comment]
I left comments in Vercel preview, but just in case they don't display correctly I'm writing what I saw, probably you already now more than half:

- The proper heading font seems to be failing and instead of defaulting to system its defaulting to very odd fonts.
- The layout for "my agreements" page is the same as the agreement page, doesn't feel right.
- The responsiveness seems to be totally broken.
- The action column looks odd when there is no action available.
- The agreement positions component don't match the design.
- If the ENS avatar fails to load (something that happens very often with mine, it breaks the mini avatar preview in the positions list.
- I'm missing the loading and "not found" state.

Tl;dr: It's very broken right now, but looks promising.

**Comment by gordo-labs** [comment]
> Probably you already now more than half:
> 
- [ X]  The proper heading font seems to be failing and instead of defaulting to system its defaulting to very odd fonts.
- [ X]  The layout for "my agreements" page is the same as the agreement page, doesn't feel right.
Finalised a stable middle point to continue works from here
- [ X]  The responsiveness seems to be totally broken.
Finalised a stable middle point to continue works from here
- [ X]  The action column looks odd when there is no action available.
Built the No Action options and final component
- [ X]  The agreement positions components don't match the design.
Fixed broken design
- [ ]  If the ENS avatar fails to load (something that happens very often with mine) it breaks the mini avatar preview in the positions list.
I need to deep check
- [ X]  I'm missing the loading and "not found" state.
Done

@xPi2 I think an outside recheck here looking for critical breaking of the flow that I'm not seeing, would help to continue.
Stopping works over this branch for now.



> Probably you already now more than half:
> 
> * The proper heading font seems to be failing and instead of defaulting to system its defaulting to very odd fonts.
> * The layout for "my agreements" page is the same as the agreement page, doesn't feel right.
> * The responsiveness seems to be totally broken.
> * The action column looks odd when there is no action available.
> * The agreement positions components don't match the design.
> * If the ENS avatar fails to load (something that happens very often with mine) it breaks the mini avatar preview in the positions list.
> * I'm missing the loading and "not found" state.
> 
> Tl;dr: Looks promising, but it's very broken right now.

File: .vscode/settings.json
Comment (xPi2) [inline .vscode/settings.json:1]: Probably we should add .vscode to gitignore

File: packages/ui-components/src/components/Atoms/typography/Headline4.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React, { ReactNode } from "react";
+
+interface Props {
+	children: ReactNode;
+	className?: string;
+}
+
+export const Headline4 = (props: Props) => {
```
Comment (xPi2) [inline packages/ui-components/src/components/Atoms/typography/Headline4.tsx:8]: Just curious, why you never use destructuring?

File: packages/ui-components/src/components/Organisms/SideNav.tsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ export const SideNav = ({
 	connectionButton,
 }: SideBarProps) => {
 	return (
-		<div className="flex flex-col w-72 h-screen items-center justify-center">
+		<div className=" flex flex-col w-72 h-screen items-center justify-center">
```
Comment (xPi2) [inline packages/ui-components/src/components/Organisms/SideNav.tsx:25]: XD

File: packages/ui-components/theme/colors.ts
Comment (xPi2) [inline packages/ui-components/theme/colors.ts:1]: A little complex naming system, don't you think? XD

File: packages/web-app/components/agreement/NotFound.tsx
Comment (xPi2) [inline packages/web-app/components/agreement/NotFound.tsx:1]: I didn't manage to trigger the Not Found view

**Review by xPi2 — COMMENTED** [review]
Tremendous work, I still need to test the preview, but code wise looks great, just minor comments on style that we can amend in later PRs

File: packages/ui-components/src/components/Organisms/SideNav.tsx (hunk 2 above)
Comment (gordo-labs) [reply packages/ui-components/src/components/Organisms/SideNav.tsx:25]: oh 😅

File: packages/ui-components/theme/colors.ts
Comment (gordo-labs) [reply packages/ui-components/theme/colors.ts:1]: Noooo, no worrries, the problem is that we carried out naming from the figma files directly.
And for the sake of continuing without this being a problem I just started building and modify the names to easy descriptive ones as I was using them, and back to the figma at the same time.

We are building as well a set of components to automate some exports, they now yet require a lot manual touch.

The rule is and will be for all
[type]-[color]-[grade]

File: packages/web-app/components/agreement/NotFound.tsx
Comment (gordo-labs) [reply packages/web-app/components/agreement/NotFound.tsx:1]: It should now be working

File: packages/ui-components/src/components/Atoms/typography/Headline4.tsx (hunk 1 above)
Comment (gordo-labs) [reply packages/ui-components/src/components/Atoms/typography/Headline4.tsx:8]: In fact that one is not correct, lol >.<
